File: ruby/lib/resp_bench/command/impl/get_command.rb
# frozen_string_literal: true

module RespBench
  module Command
    module Impl
      # GET command implementation.
      class GetCommand < Command
        def execute(client, key_generator)
          key = key_generator.next_key
          result = client.get(key)
          CommandResult.new(
            command_name: @name,
            latency_micros: result.latency_micros,
            success: result.success?
          )
        end
      end
    end
  end
end

File: python/src/resp_bench/command/impl/get_command.py
"""GET command."""

from __future__ import annotations

from typing import Optional

from ...client.benchmark_client import AsyncBenchmarkClient
from ..command import Command, CommandResult


class GetCommand(Command):
    async def execute(self, client: AsyncBenchmarkClient, key: Optional[str]) -> CommandResult:
        result = await client.get(key)
        return CommandResult(
            command_name=self.name,
            latency_micros=result.latency_micros,
            success=result.success,
        )
